### DISCUSSION TITLE
Release Process?

### DISCUSSION THREAD

**Discussion by LoneRifle** [node-saml/xml-crypto#314] [General]
I'm aware that thus far, @cjbarth has a process for pushing out more recent releases of this package. Is this documented anywhere? Would be helpful should others need to step in when he's away.

**Comment by cjbarth** [comment]
`npm run release`. After the changelog gets updated check it for accuracy before following the prompt to commit it. I usually just have to change `master` to the version number that I'm releasing. It is really just that simple. I'll document it here next time I do it.

**Reply by LoneRifle** [reply]
Would you think it would be helpful to automate this on GitHub Actions? We could trigger a release process when a GitHub release is created from a commit-ish, which can then run `npm run release` and configure it to avoid prompting one way or another

**Reply by cjbarth** [reply]
It would require putting secrets to NPM up there. I'm not entirely sure I feel comfortable with that. I don't know much about how handling those secrets would work. Do you have experience in that? It would also require some doing to make sure that we handle the changelog stuff correctly. It is nice to have a final set of eyes on the changelog before we push anything. I've caught several things that way and aborted the release.

**Reply by LoneRifle** [reply]
I've done it with a few of the repos at @opengovsg . If you still need a human in the loop to check the changelog then we can hold off on the automation. That said, it might suggest something in the process that can be made more mechanical, or at least documented with guidelines describing what would abort the release.
